financial-connections/src/main/java/com/stripe/android/financialconnections/navigation/NavigationManager.kt

```diff
@@ -1,59 +1,67 @@
+    data class NavigateBack(
+        val route: String? = null,
+        val inclusive: Boolean = false,
+    ) : NavigationIntent()
 
-    override val navigationState: MutableStateFlow<NavigationState> =
-        MutableStateFlow(NavigationState.Idle)
+    data class NavigateTo(
+        val route: String,
+        val popUpToCurrent: Boolean = false,
+        val inclusive: Boolean = false,
+        val isSingleTop: Boolean = true,
+    ) : NavigationIntent()
+}
 
-    override fun navigate(state: NavigationState) {
-        logger.debug("NavigationManager navigating to: $navigationState")
-        navigationState.value = state
-    }
+internal class NavigationManagerImpl @Inject constructor() : NavigationManager {
+    override val navigationChannel = Channel<NavigationIntent>(
+        capacity = Int.MAX_VALUE,
+        onBufferOverflow = BufferOverflow.DROP_LATEST,
+    )
 
-    override fun onNavigated(state: NavigationState) {
-        // clear navigation state, if state is the current state:
-        navigationState.compareAndSet(state, NavigationState.Idle)
+    override fun tryNavigateBack(route: String?, inclusive: Boolean) {
+        navigationChannel.trySend(
+            NavigationIntent.NavigateBack(
+                route = route,
+                inclusive = inclusive
+            )
+        )
     }
-}
 
-@Suppress("ComplexMethod")
-internal fun Pane.toNavigationCommand(
-    args: Map<String, Any?> = emptyMap()
-): NavigationCommand = when (this) {
-    Pane.INSTITUTION_PICKER -> NavigationDirections.institutionPicker
-    Pane.PARTNER_AUTH -> NavigationDirections.partnerAuth
-    Pane.CONSENT -> NavigationDirections.consent
-    Pane.ACCOUNT_PICKER -> NavigationDirections.accountPicker
-    Pane.SUCCESS -> NavigationDirections.success
-    Pane.MANUAL_ENTRY -> NavigationDirections.manualEntry
-    Pane.MANUAL_ENTRY_SUCCESS ->
-        NavigationDirections.ManualEntrySuccess(args)
-    Pane.ATTACH_LINKED_PAYMENT_ACCOUNT -> NavigationDirections.attachLinkedPaymentAccount
-    Pane.RESET -> NavigationDirections.reset
-    Pane.NETWORKING_LINK_SIGNUP_PANE -> NavigationDirections.networkingLinkSignup
-    Pane.NETWORKING_LINK_LOGIN_WARMUP -> NavigationDirections.networkingLinkLoginWarmup
-    Pane.NETWORKING_LINK_VERIFICATION -> NavigationDirections.networkingLinkVerification
-    Pane.LINK_STEP_UP_VERIFICATION -> NavigationDirections.linkStepUpVerification
-    Pane.LINK_ACCOUNT_PICKER -> NavigationDirections.linkAccountPicker
-    Pane.NETWORKING_SAVE_TO_LINK_VERIFICATION -> NavigationDirections.networkingSaveToLinkVerification
-    Pane.BANK_AUTH_REPAIR,
-    Pane.AUTH_OPTIONS,
-    Pane.LINK_CONSENT,
-    Pane.LINK_LOGIN,
-    Pane.UNEXPECTED_ERROR -> {
-        TODO("Unimplemented navigation command: ${this.value}")
+    override fun tryNavigateTo(
+        route: String,
+        popUpToCurrent: Boolean,
+        inclusive: Boolean,
+        isSingleTop: Boolean
+    ) {
+        navigationChannel.trySend(
```

Comment:
Actually moved to a `SharedFlow` since we don't want to process multiple events: https://github.com/stripe/stripe-android/pull/7250/commits/3461ef733b95ed383d2bc96234a03abbad6e98b6